System/Posix/Directory/Common.hsc

```diff
@@ -345,12 +321,8 @@ foreign import ccall unsafe "__hscore_d_type"
   d_type :: Ptr CDirent -> IO CChar
 
 -- traversing directories
-foreign import ccall unsafe "__hscore_readdir"
-  c_readdir  :: Ptr CDir -> Ptr (Ptr CDirent) -> IO CInt
-
-foreign import ccall unsafe "__hscore_free_dirent"
-  c_freeDirEnt  :: Ptr CDirent -> IO ()
-
+foreign import ccall unsafe "readdir"
```

Comment:
Are you sure this works on mac? We've had horrible bugs before when not using `capi`.

Reply:
Changing that to `capi` fails locally with

```
/tmp/ghc233912_0/ghc_160.c:7:102: error:
     error: unknown type name ‘DIR’
        7 | struct dirent* ghczuwrapperZC0ZCunixzm2zi8zi7zi0zminplaceZCSystemziPosixziDirectoryziCommonZCreaddir(DIR* a1) {return readdir(a1);}
          |
```

Busy with work so not looking into it right now.